### ISSUE TITLE
ssh_config is not parsed correctly with recent options

### ISSUE THREAD

**Issue by omgold** [hercules-team/augeas#272]
I had two problems with the ssh (/etc/ssh_config) lense. At first it didn't accept indents on lines for Ciphers, MACs, etc., second it didn't split by comma the new options HostKeyAlgorithms and KexAlgorithms. I created a patch to fix that.

**Comment by omgold** [comment]
Here is the fix:

--- /usr/share/augeas/lenses/dist/ssh.aug	2015-07-31 10:41:00.244537819 +0200
+++ ssh.aug	2015-07-31 10:43:22.666885846 +0200
@@ -51,7 +51,7 @@
 
     let commas_entry (k:regexp) =
          let value = [ seq "commas_entry" . value_to_comma]
-      in [ key k . counter "commas_entry" . spc .
+      in [ indent . key k . counter "commas_entry" . spc .
            Build.opt_list value comma . eol ]
 
     let fw_entry (k:regexp) = [ indent . key k . spc .
@@ -66,6 +66,8 @@
 
     let ciphers = commas_entry /Ciphers/i
     let macs	= commas_entry /MACs/i
+    let hostkeyalgorithms = commas_entry /HostKeyAlgorithms/i
+    let kexalgorithms	= commas_entry /KexAlgorithms/i
 
     let special_entry = send_env
 	                    | proxy_command
@@ -73,6 +75,8 @@
 	                    | local_fw
 	                    | macs
 	                    | ciphers
+	                    | hostkeyalgorithms
+	                    | kexalgorithms
 
     let key_re = /[A-Za-z0-9]+/
                - /SendEnv|Host|ProxyCommand|RemoteForward|LocalForward|MACs|Ciphers/i

**Comment by domcleal** [comment]
Could you rebase your patch against master and open a pull request please?  It looks to be against an old version, as master currently does parse KexAlgorithms as a list, but not HostKeyAlgorithms.

**Comment by omgold** [comment]
Can't see KexAlgorithms in master, but I will do the rebase.

**Comment by domcleal** [comment]
https://github.com/hercules-team/augeas/blob/master/lenses/sshd.aug#L115

**Comment by domcleal** [comment]
Ah, Ssh versus Sshd modules, apologies.

**Comment by raphink** [comment]
This patch would be more efficient as one entry:

```augeas
let algorithms = commas_entry /(Kex|HostKey)Algorithms/i
```

**Comment by raphink** [comment]
Indentation for `commas_entry` has been supported since 1b1c3d91 and released in 1.4.0.

**Comment by omgold** [comment]
Yes, the part about the indention is correct, thus my pull request doesn't include it anymore.

**Comment by raphink** [comment]
@omgold you have a pull request associated to this issue?

**Comment by raphink** [comment]
Ah I see, https://github.com/hercules-team/augeas/pull/273

**Comment by raphink** [comment]
Can you just address the comments I added to your pull request a month ago and rebase then?
